WIP: Release Test Kitchen 2.0 (#1532)
test-kitchen/test-kitchen: RELEASE_NOTES.md

# Test Kitchen 2.0 Release Notes

> Test Kitchen 2.0 is primarily a code cleanup release, but as it does remove functionality that a limited number of users may have depended on we have chosen to release this as a new major version. We believe that nearly all users will experience no change with this new version, but it allowed us to remove a good amount of legacy logic that will make Test Kitchen a bit faster and certainly easier to maintain going forward.

Reviewer: ```suggestion
Test Kitchen 2.0 is primarily a code cleanup release, but as it does remove functionality upon a limited number of users may have depended, we have chosen to release this as a new major version. We believe that nearly all users will experience no change with this new version, but this release allows us to remove a good amount of legacy logic that will make Test Kitchen a bit faster and certainly easier to maintain going forward.
```

Reviewer: Would `version` rather than `release` be a better word choice for the second sentence?

> Test Kitchen 2.0 is primarily a code cleanup release, but as it does remove functionality upon a limited number of users may have depended, we have chosen to release this as a new major version. We believe that nearly all users will experience no change with this new version, but this release allows us to remove a good amount of legacy logic that will make Test Kitchen a bit faster and certainly easier to maintain going forward.

Reviewer: ```suggestion
Test Kitchen 2.0 is primarily a code cleanup release, but as it does remove functionality that a limited number of users may have depended upon, we have chosen to release this as a new major version. We believe that nearly all users will experience no change with this new version, but this release allows us to remove a good amount of legacy logic that will make Test Kitchen a bit faster and certainly easier to maintain going forward.
```

Reviewer: Sorry. I had changed my mind about this structure.

### Breaking Changes

> - Suuport for the Chef Librarian depsolver has been removed. Users should utilize either Berkshelf or Policyfiles instead.

Reviewer: ```suggestion
- Suport for the Chef Librarian depsolver has been removed. Users should utilize either Berkshelf or Policyfiles instead.
```

- The Chef Solo provisioner no longer supports Chef 10.
- The Chef-Zero provisioner no longer supports Chef versions prior to 11.8.

### Updated SSH Support

Test Kitchen has been updated to support net-ssh 5, which allows for new SSH key formats. Thanks [@Val](https://github.com/Val) for this fix.

### Other Improvements

- Improvements to log messages better convey what Test Kitchen is actually doing and how to resolve problems

> - The kitchen init command now generates a kitchen.yml with an Ubuntu 18.04 system

Reviewer: ```suggestion
- The `kitchen init` command now generates a `kitchen.yml` with an Ubuntu 18.04 system
```

Reviewer: Correct me if I'm wrong about the tick mark formatting here.

Author: That should make sense

# Test Kitchen 1.24.0 Release Notes

## Improved error messages

> Test Kitchen now provides improved error messages when it can't find a driver, provisioner, transport, or verifier. It will now input a message listing the plugins it does know about, which may help users that have simply typod a name in a kitchen.yml file

Reviewer: ```suggestion
Test Kitchen now provides improved error messages when it can't find a driver, provisioner, transport, or verifier. It will now input a message listing the plugins it does know about, which may help users who have simply typed a name in a kitchen.yml file
```

> Test Kitchen now provides improved error messages when it can't find a driver, provisioner, transport, or verifier. It will now output a message listing the plugins that it does know about, which may help users who have simply typed a name in a kitchen.yml file

Reviewer: ```suggestion
Test Kitchen now provides improved error messages when it can't find a driver, provisioner, transport, or verifier. It will now output a message listing the plugins that it does know about, which may help users who have simply typed a name in a `kitchen.yml` file
```

Reviewer: Trying to be consistent in formatting of the release notes here.

Reviewer: Also, `can't` should be `cannot`.

Allows a user to use data_bags from an array of directories

> ```

Reviewer: SO, in line 70, a ``` needs to be added to the beginning of " KITCHEN_LOCAL_YML`" so the rest of the release notes formatted correctly. :)

Reviewer: Sorry - line 71!
